woocommerce/woocommerce
# Cleanup task list and re-organize file structure

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



While [updating the task list document](https://github.com/woocommerce/woocommerce/pull/38245), I noticed that we have many outdated code that can be removed or deprecated, including task snooze APIs and old two-column experiment code.

Besides, we have three folders related to the task lists, which is very confusing. 😕 

- [`tasks`](https://github.com/woocommerce/woocommerce/tree/7e3729c5fcee710a6bb79bde4291577fbef14aa2/plugins/woocommerce-admin/client/tasks): include task list entry file
- [`task-lists`](https://github.com/woocommerce/woocommerce/tree/7e3729c5fcee710a6bb79bde4291577fbef14aa2/plugins/woocommerce-admin/client/task-lists): only contains progress title and header
- [`two-column-tasks`](https://github.com/woocommerce/woocommerce/tree/7e3729c5fcee710a6bb79bde4291577fbef14aa2/plugins/woocommerce-admin/client/two-column-tasks):  It's acutally setup task list, not related to two-columns.

Thus, this PR makes the following changes:

- Deprecate onboarding task snooze APIs c30d63c59a8fc68f95f8c6c22d990ee723cde0a2 (snooze feature has been drepcated in https://github.com/woocommerce/woocommerce/pull/35450)
- Removes the remaining `two-column` experiment code in the task list (in both backend and frontend)
- Re-organizes task list file structure:
   - Merge `./client/tasks` folder into `./client/task-lists` To express what it actually is.
   - Rename  [`two-column-tasks`](https://github.com/woocommerce/woocommerce/tree/7e3729c5fcee710a6bb79bde4291577fbef14aa2/plugins/woocommerce-admin/client/two-column-tasks) to `setup-task-list` and also move it into `./client/task-lists` folder
   - Update the corresponding component names and import paths
   - Moves components to `./task-lists/components` folder
  
After merging this, we can find all frontend task list files in [./client/task-lists/](https://github.com/woocommerce/woocommerce/tree/dev/clean-up-task-list/plugins/woocommerce-admin/client/task-lists) directory.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate WooCommerce in a brand new site
2. Go to `WooCommerce > Home`
3. Confirm the task lists are shown as expected as previous 
4. Perform a smoke test on the task list
5. Complete set up a task list
6. Observe that task list completion UI is shown
